### PR TITLE
Relocate resolveNSGReference in Azure node driver

### PR DESCRIPTION
**Pull request by aiyengar2**
Related to a bug introduced in https://github.com/rancher/machine/pull/77, detailed in the issue.

Related Issue: https://github.com/rancher/rancher/issues/27468

### CODE REVIEW

**Comment by aiyengar2** [comment]
TLDR for why this is required: SetConfigFlags is only called on Create and not on Remove.

Rancher interacts with machine as an executable, so after pulling up all the necessary flags from GetCreateFlags and storing those fields within the NodeDriver struct, any step that requires provisioning (i.e. the creation of a new node) triggers a rancher-machine call like:

```
rancher-machine create <flags>
```

Internally, this triggers the creation step at the following:
https://github.com/rancher/machine/blob/d32ea31a178bb4fe952f1885666aafa30850400c/commands/commands.go#L221-L226

Defined within `command/create.go`, `cmdCreateOuter` calls `addDriverFlagsToCommand`, which in turn calls `cmdCreateInner`. This is where `SetConfigFlags` is called:
https://github.com/rancher/machine/blob/d32ea31a178bb4fe952f1885666aafa30850400c/commands/create.go#L222-L224

However, it is not called on Remove (since the only function for which it is ever called is on Create), so the Config is never reset.

**Comment by aiyengar2** [comment]
> These should both use the same style

Good catch. Fixed in the latest commit!
